File: 278. First Bad Version/Solution.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);
// 1660450398 
class Solution {
public:
    int firstBadVersion(int n) {
        int start=1,end=n;
        while(start<end){
            int mid = start+(end-start)/2;
            if(!isBadVersion(mid)){
                start = mid + 1;
            }else{
                end = mid;
            }
        }
        return end;
    }
};

File: 278. First Bad Version/Solution.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:
# 1660452318
class Solution:
    def firstBadVersion(self, n: int) -> int:
        start,end = 1,n
        while start<end:
            mid = (start+end)//2
            if not isBadVersion(mid):
                start = mid + 1
            else:
                end = mid
        return end
